pandas/core/generic.py

```diff
@@ -5304,7 +5304,11 @@ def sample(
                         "when sampling from a Series."
                     )
 
-            weights = pd.Series(weights, dtype="float64")
+            if isinstance(self, ABCSeries):
+                func = self._constructor
+            else:
+                func = self._constructor_sliced
```

Comment:
Or is it better to import `Series` within this function so as to avoid an if-else block?

Reply:
yeah you can just import Series here

Reply:
the only reason you woul want this is to handle a subclass

Reply:
but i guess this is ok